receiver/azureeventhubreceiver/azure_formatter_impl.go

```diff
@@ -0,0 +1,237 @@
+// instances with a raw int or double. This function
+// returns the possibly mutated interface.
+func replaceJsonNumber(i interface{}) interface{} {
+	switch t := i.(type) {
+	case map[string]interface{}:
+		for k, v := range t {
+			switch number := v.(type) {
+			case json.Number:
+				t[k] = jsonNumberToRaw(number)
+			default:
+				replaceJsonNumber(v)
+			}
+		}
+	case []interface{}:
+		for k, v := range t {
+			switch number := v.(type) {
+			case json.Number:
+				t[k] = jsonNumberToRaw(number)
+			default:
+				replaceJsonNumber(v)
+			}
+		}
+	}
+	return i
+}
+
+// extractRawAttributes creates a raw attribute map and
+// inserts attributes from the Azure log record. Optional
+// attributes are only inserted if they are defined. The
+// azureDuration value is only set if the value in the
+// Azure log record can be parsed as an integer.
+func extractRawAttributes(log azureLogRecord) map[string]interface{} {
+	var attrs = map[string]interface{}{}
+
+	attrs[azureCategory] = log.Category
+	setIf(attrs, azureCorrelationID, log.CorrelationID)
+	if log.DurationMs != nil {
+		duration, err := strconv.ParseInt(*log.DurationMs, 10, 64)
+		if err == nil {
+			attrs[azureDuration] = duration
+		}
+	}
+	if log.Identity != nil {
+		attrs[azureIdentity] = replaceJsonNumber(*log.Identity)
+	}
+	attrs[azureOperationName] = log.OperationName
+	setIf(attrs, azureOperationVersion, log.OperationVersion)
+	if log.Properties != nil {
+		attrs[azureProperties] = replaceJsonNumber(*log.Properties)
+	}
+	setIf(attrs, azureResultDescription, log.ResultDescription)
+	setIf(attrs, azureResultSignature, log.ResultSignature)
+	setIf(attrs, azureResultType, log.ResultType)
+	setIf(attrs, azureTenantID, log.TenantID)
+
+	setIf(attrs, cloudRegion, log.Location)
+	attrs[cloudProvider] = "azure"
+
+	setIf(attrs, netSockPeerAddr, log.CallerIPAddress)
+
+	return attrs
+}
+
+// transform takes a byte array containing a JSON-encoded
+// payload with Azure log records and transforms it into
+// an OpenTelemetry plog.Logs object. The data in the Azure
+// log record appears as fields and attributes in the
+// OpenTelemetry representation; the bodies of the
+// OpenTelemetry log records are empty.
+func transform(data []byte) (*plog.Logs, error) {
+
+	l := plog.NewLogs()
+
+	var azureLogs azureRecords
+	decoder := json.NewDecoder(bytes.NewReader(data))
+	decoder.UseNumber()
+	err := decoder.Decode(&azureLogs)
+	resourceLogs := l.ResourceLogs().AppendEmpty()
+	if err == nil {
+		logRecords := resourceLogs.ScopeLogs().AppendEmpty().LogRecords()
+
+		resourceID := ""
+		for _, azureLog := range azureLogs.Records {
+			resourceID = azureLog.ResourceID
+			nanos, err := asTimestamp(azureLog.Time)
+			if err == nil {
+				lr := logRecords.AppendEmpty()
+
+				lr.SetTimestamp(nanos)
+
+				if azureLog.Level != nil {
+					severity := asSeverity(*azureLog.Level)
+					lr.SetSeverityNumber(severity)
+					lr.SetSeverityText(*azureLog.Level)
+				}
+
+				//nolint:errcheck
```

Comment:
Probably a good idea to capture and return early in case of errors, wdyt?

Reply:
This error should never happen because the raw map is constructed from valid JSON, matching the Azure common log format. However, I can capture the last error and return it. However, with the processing at the next level up, this would mean that *all* logs in this message would be ignored. I'm not sure that's the right behavior. However, given that this should never happen

Reply:
It should never happen, but then someone else might modify the code down the road and create problems. Just remove any uncertainty if possible.

Reply:
@atoulme I'll return early for errors here. Do you want the same behavior in the "raw" formatter as well? The equivalent error is also ignored there.

Reply:
yes, please

Reply:
Done.